source/patterns/@aws-solutions-constructs/core/test/kendra-helper.test.ts

```diff
@@ -0,0 +1,555 @@
+    roleName: 'externalFakeRole'
+  });
+
+  // Create data source props
+  const otherDataSourceProps: Partial<kendra.CfnDataSourceProps> = {
+    name: nonImplementedSourceName,
+    roleArn: fakeRole.roleArn,
+    type: nonImplementedSourceType
+  };
+
+  // Add a data source
+  const sourceObjects = AddMultipleKendraDataSources(stack, 'test-other-source', newIndex, [
+    s3DataSourceProps,
+    otherDataSourceProps
+  ]
+  );
+
+  expect(sourceObjects.length).toEqual(2);
+  expect(sourceObjects[0].type).toEqual('S3');
+  expect(sourceObjects[1].name).toEqual(nonImplementedSourceName);
+
+  const template = Template.fromStack(stack);
+  template.resourceCountIs('AWS::Kendra::Index', 1);
+  template.resourceCountIs('AWS::Kendra::DataSource', 2);
+  template.resourceCountIs('AWS::IAM::Role', 3);
+
+  // Look for S3 data source
+  template.hasResourceProperties("AWS::Kendra::DataSource", {
+    Name: testName,
+    RoleArn: Match.anyValue(),
+    Type: 'S3',
+    DataSourceConfiguration: {
+      S3Configuration: {
+        BucketName: testBucketName
+      }
+    }
+  });
+  template.hasResourceProperties("AWS::IAM::Role", {
+    AssumeRolePolicyDocument: {
+      Statement: [
+        {
+          Action: "sts:AssumeRole",
+          Effect: "Allow",
+          Principal: {
+            Service: "kendra.amazonaws.com"
+          }
+        }
+      ],
+      Version: "2012-10-17"
+    },
+    Policies: [
+      {
+        PolicyDocument: {
+          Statement: [
+            {
+              Action: "s3:GetObject",
+              Effect: "Allow",
+              Resource: `arn:aws:s3:::${testBucketName}/*`
+            },
+            {
+              Action: "s3:ListBucket",
+              Effect: "Allow",
+              Resource: `arn:aws:s3:::${testBucketName}`
+            },
+            {
+              Action: [
+                "kendra:BatchPutDocument",
+                "kendra:BatchDeleteDocument"
+              ],
+              Effect: "Allow",
+              Resource: {
+                "Fn::GetAtt": [
+                  "kendraindextest",
+                  "Arn"
+                ]
+              }
+            }
+          ],
+          Version: "2012-10-17"
+        },
+        PolicyName: "s3CrawlPolicy"
+      }
+    ]
+  });
+
+  // Look for non-implemented data source
+  template.hasResourceProperties("AWS::Kendra::DataSource", {
+    Name: nonImplementedSourceName,
+    Type: nonImplementedSourceType,
+    IndexId: Match.anyValue()
+  });
+
+});
+
+test('Confirm Errof for bad kendra Permission in normalizeKendraPermissions()', () => {
+  const inputs = ["read", "submitfeedback", "write"];
+
+  const outputs = normalizeKendraPermissions(inputs);
+
+  expect(outputs.includes("READ")).toBeTruthy();
+  expect(outputs.includes("SUBMITFEEDBACK")).toBeTruthy();
+  expect(outputs.includes("WRITE")).toBeTruthy();
+});
+
+test('Confirm successful operation of normalizeKendraPermissions()', () => {
```

Comment:
Glancing at the name of this test, and then the assertion that it threw an error made me do a double take, assuming either the name was misleading or the test was inaccurate. After studying it for a moment, it makes sense, but I wonder if it would be more clear to reword the test along the lines of `Confirm bad kendra index values throws exception` or similar.